src/test/scala/zio/interop/JavaSpec.scala

```diff
@@ -15,7 +15,7 @@ object JavaSpec {
       testM("be lazy on the `Future` parameter") {
         var evaluated         = false
         def ftr: Future[Unit] = CompletableFuture.supplyAsync(() => evaluated = true)
-        assertM(ZIO.fromFutureJava(UIO.effectTotal(ftr)).as(evaluated).run, succeeds(isFalse))
+        assertM(ZIO.fromFutureJava(UIO.effectTotal(ftr)).when(false).as(evaluated), isFalse)
```

Comment:
In RC14 `as` become lazy on it's param, IMO uncovering race condition in test.
`ftr` was run, but `as(evaluated)` is computed as `false` during construction of effect and `ftr` is run when executing effect.